### PR TITLE
chore: merge #17623 to main

### PR DESCRIPTION
**Pull request by hpidcock**
Forward ports #17623 to main

Conflicts:
- api/controller/migrationmaster/client.go
- api/controller/migrationmaster/client_test.go
- apiserver/facades/client/controller/controller.go
- apiserver/facades/client/controller/controller_test.go
- apiserver/facades/client/controller/export_test.go
- apiserver/facades/client/controller/register.go
- apiserver/facades/controller/migrationmaster/facade.go
- apiserver/facades/controller/migrationmaster/facade_test.go

### CODE REVIEW

File: apiserver/facades/client/controller/controller.go [hunk 1]
```diff
@@ -852,7 +880,7 @@ func (c *ControllerAPI) ConfigSet(ctx context.Context, args params.ControllerCon
 	return nil
 }
 
-// runMigrationPrechecks runs prechecks on the migration and updates
+// runMigrationPreChecks runs prechecks on the migration and updates
```
Comment (wallyworld) [inline apiserver/facades/client/controller/controller.go:883]: typo, original was correct

**Review by wallyworld — APPROVED** [review]

File: apiserver/facades/client/controller/controller.go (hunk 1 above)
Comment (hpidcock) [reply apiserver/facades/client/controller/controller.go:883]: Will fix later.

**Pull request merged**